src/controllers/guildRoleHandler.ts

```diff
@@ -68,3 +70,53 @@ export async function removeGuildRoleHandler(request: IRequest, env: env) {
     });
   }
 }
+export async function getGuildRolesHandler(request: IRequest, env: env) {
+  const authHeader = request.headers.get("Authorization");
+
+  if (!authHeader) {
+    return new JSONResponse(response.BAD_SIGNATURE, { status: 401 });
+  }
+  try {
+    await verifyAuthToken(authHeader, env);
+    const roles = await getGuildRoles(env);
+    return new JSONResponse({ roles });
+  } catch (err: any) {
+    if (err.message === response.ROLE_FETCH_FAILED_MESSAGE) {
+      return new JSONResponse(response.ROLE_FETCH_FAILED_ERROR, {
+        status: 500,
+      });
+    }
+    return new JSONResponse(response.INTERNAL_SERVER_ERROR, { status: 500 });
+  }
+}
+
+export async function getGuildRoleByRoleNameHandler(
+  request: IRequest,
+  env: env
+) {
+  const authHeader = request.headers.get("Authorization");
+  const roleName = decodeURI(request.params?.roleName ?? "");
+
+  if (!authHeader) {
+    return new JSONResponse(response.BAD_SIGNATURE, { status: 401 });
+  }
+
+  if (!roleName) {
+    return new JSONResponse(response.NOT_FOUND, { status: 404 });
+  }
+  try {
+    await verifyAuthToken(authHeader, env);
+    const role = await getGuildRoleByName(roleName, env);
+    if (!role) {
+      return new JSONResponse(response.NOT_FOUND, { status: 404 });
+    }
+    return new JSONResponse(role);
+  } catch (err: any) {
+    if (err.message === response.ROLE_FETCH_FAILED_MESSAGE) {
+      return new JSONResponse(response.ROLE_FETCH_FAILED_ERROR, {
+        status: 500,
+      });
+    }
+    return new JSONResponse(response.INTERNAL_SERVER_ERROR, { status: 500 });
```

Comment:
we need to add a header as well in case we are passing a status code.

Reply:
Yes, understood. 
Since I am overwriting the init object, 
Thanks for pointing this out as well. 🙏

I have fixed it in the new commit.